Remotion/Web/UnitTests/Core/UI/Controls/WebButtonTests/EncodingTest.cs

```diff
@@ -0,0 +1,81 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System.IO;
+using System.Web.UI;
+using NUnit.Framework;
+using Remotion.Development.Web.UnitTesting.UI.Controls;
+
+namespace Remotion.Web.UnitTests.Core.UI.Controls.WebButtonTests
+{
+  [TestFixture]
+  public class EncodingTest : BaseTest
+  {
+    private TestWebButton _webButton;
+
+    protected override void SetUpPage()
+    {
+      base.SetUpPage();
+      _webButton = new TestWebButton();
+      _webButton.ID = "WebButton";
+    }
+
+    [Test]
+    public void Text_WithPlainText ()
+    {
+      _webButton.Text = WebString.CreateFromText ("test");
+      var result = RenderControl (_webButton);
+
+      Assert.That (result, Does.Contain (@"<span class=""buttonBody""><span>test</span></span>"));
```

Comment:
We have a control test infrastructure. See BoctextValueRendererTest etc. Removes details such as insignificant whilespace and attribute ordering. Please update.

Setup: Html = new HtmlHelper();

Test:
_webButton.Render (Html.Writer);
var document = Html.GetResultDocument();
Html.AssertChildElementCount (document.DocumentElement, 1);
var span = Html.GetAssertedChildElement (document, "button", 0);
Html.AssertAttribute (span, "class", "buttonBody");

etc etc